legend-pure-code-compiled-core-persistence/src/main/resources/core_persistence/external/format/awspersistence/transformation.pure

```diff
@@ -0,0 +1,83 @@
+import meta::pure::persistence::metamodel::batch::targetspecification::*;
+import meta::pure::persistence::metamodel::batch::deduplication::*;
+import meta::pure::persistence::metamodel::batch::mode::snapshot::*;
+import meta::pure::persistence::metamodel::batch::audit::*;
+
+
+function  meta::external::format::awspersistence::generation::transform(input: AwsPersistenceConfig[1]): AwsPersistenceOutput[*]
+{
+  meta::legend::test::mayExecuteLegendTest(
+       {clientVersion, serverVersion, serializationKind, host, port|
+                   let fStr = 'meta::protocols::pure::'+$clientVersion+'::invocation::generation::awspersistence::legendGenerateAwsPersistence_AwsPersistenceConfig_1__String_1__Integer_1__String_1__AwsPersistenceOutput_MANY_';
+                   let xf = $fStr->pathToElement()->cast(@Function<{AwsPersistenceConfig[1], String[1], Integer[1], String[1]->AwsPersistenceOutput[*]}>);
+                   $xf->evaluate([$input, $host, $port, $serverVersion]->map(v|list($v)))->toOneMany()->cast(@AwsPersistenceOutput);},
+       | meta::external::format::awspersistence::generation::internal_transform($input)
+    )->toOneMany();   
+}
+
+function <<access.private,Generation.Transformation>> meta::external::format::awspersistence::generation::internal_transform(input: AwsPersistenceConfig[1]): AwsPersistenceOutput[*]
+{
+   let possibleElement = $input.allPackageScopeElements()->filter(p|$p->instanceOf(Persistence))->cast(@Persistence);
+   assertNotEmpty($possibleElement, | 'No Persistence was found');
+   assertFalse($possibleElement->size()>1, | 'More than one Persistence was found');
+   $input->meta::external::format::awspersistence::generation::generateAwsPersistenceFromPure($possibleElement->toOne()->cast(@Persistence));
+}
+
+
+function meta::external::format::awspersistence::generation::generateAwsPersistenceFromPureWithScope(config: AwsPersistenceConfig[1]): AwsPersistenceOutput[*]
+{
+   let scopeElements = $config.allPackageScopeElements()->filter(p|$p->instanceOf(Persistence))->cast(@Persistence);
+   assertNotEmpty($scopeElements, 'At least one Persistence is needed to transform into cloud-persistent backend form');
+   $scopeElements->map(p| $config->meta::external::format::awspersistence::generation::generateAwsPersistenceFromPure($p));
+}
+
+function meta::external::format::awspersistence::generation::generateAwsPersistenceFromPure(config: AwsPersistenceConfig[1], p: Persistence[1]): AwsPersistenceOutput[*]
+{
+   let content = $config->generateAwsPersistenceBackend($p);
+   let out = ^AwsPersistenceOutput(content=$content,fileName=$p->elementToPath()->pathToFileName('awspersistence'), format='json');
+   $out;
+}
+
+function meta::external::format::awspersistence::generation::generateAwsPersistenceBackend(config: AwsPersistenceConfig[1], p: Persistence[1]): String[1]
+{
+  $config->meta::external::format::awspersistence::generation::generateAwsPersistenceBackendRaw($p)->toCompactJSONString();
```

Comment:
I found that using pretty json rather than compact could help user introspect the generated file on Studio.

Reply:
Thanks, fixed